tfhe/src/core_crypto/algorithms/lwe_keyswitch.rs

```diff
@@ -479,4 +504,18 @@ pub fn par_keyswitch_lwe_ciphertext_with_thread_count<Scalar, KSKCont, InputCont
     let reduced_ksed_body = *reduced.get_body().data;
     // Copy the input body to the output ciphertext
     *output_lwe_ciphertext.get_mut_body().data = input_lwe_body.wrapping_add(reduced_ksed_body);
+
+    // If the moduli are not the same, we need to round the body in the output ciphertext
+    if output_ciphertext_modulus != input_ciphertext_modulus
+        && !output_ciphertext_modulus.is_native_modulus()
+    {
+        let modulus_bits = output_ciphertext_modulus.get_custom_modulus().ilog2() as usize;
+        let output_decomposer = SignedDecomposer::new(
+            DecompositionBaseLog(modulus_bits),
+            DecompositionLevelCount(1),
+        );
+
+        *output_lwe_ciphertext.get_mut_body().data =
+            output_decomposer.closest_representable(*output_lwe_ciphertext.get_mut_body().data);
+    }
```

Comment:
Shouldn't this be done before the keyswitch, right after `*output_lwe_ciphertext.get_mut_body().data = *input_lwe_ciphertext.get_body().data;`

Reply:
I think the 32 bits test runs the sequential and parallel code and outputs are bit exact equal, as we are playing with integers everything works ok

Reply:
I'll check that my test actually does that

Reply:
Better to do it in the same place for both versions, even if it's equivalent, don't you think?

Reply:
yes but the parallel thing allocates the output ciphertext itself IIRC, I'll see to tweak it

Reply:
yep the test runs sequential and parallel keyswitch with the same outputs and checks they are equal

Reply:
I'll see if the code can be made a bit more consistent

Reply:
turns out I overcomplicated that part, must have been in between two versions of the code and commited it as is, cleaning up